plugins/maven-enforcer-pattern-size/src/main/java/org/wildfly/swarm/plugin/enforcer/patternsize/RequireFilePatternSize.java

```diff
@@ -0,0 +1,158 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.wildfly.swarm.plugin.enforcer.patternsize;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.FileVisitOption;
+import java.nio.file.FileVisitResult;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.SimpleFileVisitor;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.util.ArrayList;
+import java.util.EnumSet;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.apache.maven.enforcer.rule.api.EnforcerRule;
+import org.apache.maven.enforcer.rule.api.EnforcerRuleException;
+import org.apache.maven.enforcer.rule.api.EnforcerRuleHelper;
+
+/**
+ *
+ * @author Juan Gonzalez
+ *
+ */
+public class RequireFilePatternSize implements EnforcerRule {
+
+    private RequiredFilePattern[] filePatterns;
+
+    public RequiredFilePattern[] getRequiredFilePatterns() {
+        return filePatterns;
+    }
+
+    public void setRequiredFilePatterns(RequiredFilePattern[] filePatterns) {
+        this.filePatterns = filePatterns;
+    }
+
+    List<RequiredFilePatternFailure> checkFilePattern(final RequiredFilePattern requiredFilePattern) {
+        if (requiredFilePattern == null || requiredFilePattern.getDirectory() == null) {
+            return null;
+        }
+
+        if (requiredFilePattern.getPattern() == null) {
+            return null;
+        }
+
+        final Pattern pattern = Pattern.compile(requiredFilePattern.getPattern());
+
+        String directory = requiredFilePattern.getDirectory();
+        File directoryFile = new File(directory);
+
+        final List<File> matchedFiles = new ArrayList<File>();
+
+        if (Files.isDirectory(directoryFile.toPath())) {
+            try {
+                Files.walkFileTree(directoryFile.toPath(), EnumSet.of(FileVisitOption.FOLLOW_LINKS),
+                    requiredFilePattern.isRecursive() ? Integer.MAX_VALUE : 1,
+                            new SimpleFileVisitor<Path>() {
+                            @Override
+                            public FileVisitResult visitFile(Path filePath, BasicFileAttributes attrs) throws IOException {
+                                 Matcher matcher = pattern.matcher(filePath.toAbsolutePath().toString());
+                                 if (matcher.matches()) {
+                                     File file = filePath.toFile();
+                                     matchedFiles.add(file);
+                                 }
+
+                                return super.visitFile(filePath, attrs);
+                            }
+                 });
+            } catch (IOException e) {
+                e.printStackTrace();
```

Comment:
I know it's just a compile-time thing, but I'd rather avoid `e.printStackTrace` completely.